diff --git a/TMessagesProj/src/main/java/org/telegram/tgnet/TLRPC.java b/TMessagesProj/src/main/java/org/telegram/tgnet/TLRPC.java
index e69f4c5..cccc43a 100644
--- a/TMessagesProj/src/main/java/org/telegram/tgnet/TLRPC.java
+++ b/TMessagesProj/src/main/java/org/telegram/tgnet/TLRPC.java
@@ -39074,7 +39074,7 @@ public class TLRPC {
             deactivated = hasFlag(flags, FLAG_5);
             call_active = hasFlag(flags, FLAG_23);
             call_not_empty = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_25);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_25))
             id = stream.readInt64(exception);
             title = stream.readString(exception);
             photo = ChatPhoto.TLdeserialize(stream, stream.readInt32(exception), exception, allowStrippedThumb);
@@ -39351,7 +39351,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -39538,7 +39538,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -39793,7 +39793,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -39976,7 +39976,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40147,7 +40147,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40310,7 +40310,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40467,7 +40467,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40616,7 +40616,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40747,7 +40747,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -40889,7 +40889,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -41027,7 +41027,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -41142,7 +41142,7 @@ public class TLRPC {
             call_not_empty = hasFlag(flags, FLAG_24);
             fake = hasFlag(flags, FLAG_25);
             gigagroup = hasFlag(flags, FLAG_26);
-            noforwards = hasFlag(flags, FLAG_27);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_27))
             join_to_send = hasFlag(flags, FLAG_28);
             join_request = hasFlag(flags, FLAG_29);
             forum = hasFlag(flags, FLAG_30);
@@ -44396,7 +44396,7 @@ public class TLRPC {
             bot_can_manage_emoji_status = hasFlag(flags2, FLAG_10);
             display_gifts_button = hasFlag(flags2, FLAG_16);
             noforwards_my_enabled = hasFlag(flags2, FLAG_23);
-            noforwards_peer_enabled = hasFlag(flags2, FLAG_24);
+            noforwards_peer_enabled = false; // plus: ignore peer content protection (was hasFlag(flags2, FLAG_24))
             unofficial_security_risk = hasFlag(flags2, FLAG_26);
             id = stream.readInt64(exception);
             if (hasFlag(flags, FLAG_1)) {
@@ -44659,7 +44659,7 @@ public class TLRPC {
             bot_can_manage_emoji_status = hasFlag(flags2, FLAG_10);
             display_gifts_button = hasFlag(flags2, FLAG_16);
             noforwards_my_enabled = hasFlag(flags2, FLAG_23);
-            noforwards_peer_enabled = hasFlag(flags2, FLAG_24);
+            noforwards_peer_enabled = false; // plus: ignore peer content protection (was hasFlag(flags2, FLAG_24))
             id = stream.readInt64(exception);
             if (hasFlag(flags, FLAG_1)) {
                 about = stream.readString(exception);
@@ -58276,7 +58276,7 @@ public class TLRPC {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
diff --git a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_ephemeral.java b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_ephemeral.java
index b065410..f7b21dc 100644
--- a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_ephemeral.java
+++ b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_ephemeral.java
@@ -108,7 +108,7 @@ public class TL_ephemeral {
             out = hasFlag(flags, FLAG_0);
             welcome = hasFlag(flags, FLAG_5);
             invert_media = hasFlag(flags, FLAG_7);
-            noforwards = hasFlag(flags, FLAG_12);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_12))
             id = stream.readInt32(exception);
             from_id = TLRPC.Peer.TLdeserialize(stream, stream.readInt32(exception), exception);
             if (hasFlag(flags, FLAG_9)) {
diff --git a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_stories.java b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_stories.java
index 9d95527..0a690df 100644
--- a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_stories.java
+++ b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/TL_stories.java
@@ -1869,7 +1869,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
@@ -1969,7 +1969,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
@@ -2063,7 +2063,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
@@ -2150,7 +2150,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
@@ -2231,7 +2231,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
@@ -2306,7 +2306,7 @@ public class TL_stories {
             isPublic = hasFlag(flags, FLAG_7);
             close_friends = hasFlag(flags, FLAG_8);
             min = hasFlag(flags, FLAG_9);
-            noforwards = hasFlag(flags, FLAG_10);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_10))
             edited = hasFlag(flags, FLAG_11);
             contacts = hasFlag(flags, FLAG_12);
             selected_contacts = hasFlag(flags, FLAG_13);
diff --git a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/legacy/TL_legacy_message.java b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/legacy/TL_legacy_message.java
index cbdf7ad..256df9f 100644
--- a/TMessagesProj/src/main/java/org/telegram/tgnet/tl/legacy/TL_legacy_message.java
+++ b/TMessagesProj/src/main/java/org/telegram/tgnet/tl/legacy/TL_legacy_message.java
@@ -26,7 +26,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -262,7 +262,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -491,7 +491,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -713,7 +713,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -929,7 +929,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -1139,7 +1139,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -1338,7 +1338,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -1531,7 +1531,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -1718,7 +1718,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -1897,7 +1897,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             flags2 = stream.readInt32(exception);
             offline = hasFlag(flags2, FLAG_1);
@@ -2070,7 +2070,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             id = stream.readInt32(exception);
             if (hasFlag(flags, FLAG_8)) {
@@ -2233,7 +2233,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             id = stream.readInt32(exception);
             if (hasFlag(flags, FLAG_8)) {
@@ -2390,7 +2390,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             id = stream.readInt32(exception);
             if (hasFlag(flags, FLAG_8)) {
@@ -2541,7 +2541,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             invert_media = hasFlag(flags, FLAG_27);
             id = stream.readInt32(exception);
             if (hasFlag(flags, FLAG_8)) {
@@ -2686,7 +2686,7 @@ public class TL_legacy_message {
             legacy = hasFlag(flags, FLAG_19);
             edit_hide = hasFlag(flags, FLAG_21);
             pinned = hasFlag(flags, FLAG_24);
-            noforwards = hasFlag(flags, FLAG_26);
+            noforwards = false; // plus: ignore content protection (was hasFlag(flags, FLAG_26))
             id = stream.readInt32(exception);
             if (hasFlag(flags, FLAG_8)) {
                 from_id = TLRPC.Peer.TLdeserialize(stream, stream.readInt32(exception), exception);
